db/types/operations/cast.py

```diff
@@ -40,6 +40,7 @@
 FULL_CHAR = base.PostgresType.CHARACTER.value
 NAME = base.PostgresType.NAME.value
 MATHESAR_CHAR = base.PostgresType.MATHESAR_CHAR.value
+MATHESAR_NAME = base.PostgresType.MATHESAR_NAME.value
```

Comment:
Same here. We shouldn't have both. We want to replace the default `"name"` implementation with ours (mainly in the `ischema_names` dict).